services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,942 @@
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not create flow: Not enough bandwidth found or path not found. " +
+                "Couldn't find non overlapping protected path"
+
+        and: "Cleanup: restore available bandwidth"
+        isls.each { database.resetIslBandwidth(it) }
+    }
+
+    def "Unable to update the 'protected path' option on a flow in case there is not enough bandwidth"() {
+        given: "Two active neighboring switches"
+        def isls = topology.getIslsForActiveSwitches()
+        def (srcSwitch, dstSwitch) = [isls.first().srcSwitch, isls.first().dstSwitch]
+
+        and: "Update all isls which can be used by protected path"
+        def bandwidth = 100
+        isls[1..-1].each { database.updateIslAvailableBandwidth(it, 90) }
+
+        when: "Create flow without protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = false
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created without protected path"
+        !northbound.getFlowPath(flow.id).protectedPath
+
+        when: "Update flow: enable protected path"
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not update flow: Not enough bandwidth found or path not found. " +
+                "Couldn't find non overlapping protected path"
+
+        and: "Cleanup: delete the flow and restore available bandwidth"
+        flowHelper.deleteFlow(flow.id)
+        isls.each { database.resetIslBandwidth(it) }
+    }
+
+    def "Able to create a flow with 'protected path' in case there is not enough bandwidth and ignoreBandwidth=true"() {
+        given: "Two active neighboring switches"
+        def isls = topology.getIslsForActiveSwitches()
+        def (srcSwitch, dstSwitch) = [isls.first().srcSwitch, isls.first().dstSwitch]
+
+        and: "Update all isls which can be used by protected path"
+        def bandwidth = 100
+        isls[1..-1].each { database.updateIslAvailableBandwidth(it, 90) }
+
+        when: "Create flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = true
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with protected path"
+        northbound.getFlowPath(flow.id).protectedPath
+
+        and: "Cleanup: delete the flow and restore available bandwidth"
+        flowHelper.deleteFlow(flow.id)
+        isls.each { database.resetIslBandwidth(it) }
+    }
+
+    @Ignore("Test is not valid, the availableBandwidth field is rewritten by each update action")
```

Comment:
If the test is not valid, why we keep it?

Reply:
removed